src/libAtomVM/opcodesswitch.h

```diff
@@ -4246,12 +4501,12 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
 
             case OP_IS_FUNCTION2: {
                 int next_off = 1;
-                int label;
-                DECODE_LABEL(label, code, i, next_off, next_off)
+                uint32_t label;
+                DECODE_LABEL(label, code, i, next_off)
                 term arg1;
-                DECODE_COMPACT_TERM(arg1, code, i, next_off, next_off)
+                DECODE_COMPACT_TERM(arg1, code, i, next_off)
                 unsigned int arity;
-                DECODE_INTEGER(arity, code, i, next_off, next_off)
+                DECODE_INTEGER(arity, code, i, next_off)
```

Comment:
Given that `DECODE_INTEGER` and `DECODE_LITERAL` have the same implementation at execution time, it is not clear to me why we replaced nearly all occurrences except for `OP_IS_FUNCTION2`

Reply:
(apparently my previous reply was lost).

Types are asserted at load time as the macros are different then, so if we put the wrong macro, e.g. `DECODE_LITERAL` here, the VM will abort. This is the whole point of this PR.

I did study coverage of every opcode for this PR and I'm pretty sure ˋOP_IS_FUNCTION2` is covered by our tests.

I can also confirm that the compiler generates an integer for ˋis_function2ˋ. I cannot access it right now but you can find an example here where the OP did paste the output of ˋerlc -Sˋ.

https://github.com/erlang/otp/issues/4092#issue-805345827

Reply:
Everything is clear now, thanks for the explanation.